misk-metrics-testing/src/main/kotlin/misk/metrics/FakeMetrics.kt

```diff
@@ -103,19 +104,22 @@ class FakeMetrics @Inject internal constructor(
   fun getSample(
     name: String,
     labels: Array<out Pair<String, String>>,
-    sampleName: String = name
+    sampleName: String? = null
   ): Sample? {
     val metricFamilySamples = registry.metricFamilySamples()
       .asSequence()
       .firstOrNull { it.name == name }
       ?: return null
 
+    val familySampleName = sampleName
+      ?: if (metricFamilySamples.type == Collector.Type.COUNTER) "${name}_total" else name
```

Comment:
Problem with the tests is that calling `metrics.counter("gets", "-", labelNames = listOf("status"))` creates a  metric family named `gets` but the sample name is `gets_total` so unless the suffix is added for counter metrics, the `FakeMetrics.get("gets"...)` cannot find the samples and `FakeMetrics.get("gets_total"...) cannot find the metric family.

This fix is not particularly pretty (the better fix may be to introduce new getters that apply the suffixes like for `histogram` values), but went this path the cause the least disruption in the library API for a smoother upgrade for consumers.

The same solution has been applied to the v2 metrics.